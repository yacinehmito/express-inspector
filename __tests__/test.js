const path = require("path");
const { tree, trace, format } = require("..");
const { takeUntil } = require("../src/utils");
const express = require("express");

trace(express);

const app = require("./_app");

function makePathRelativeToProject(absolutePath) {
  return path.relative(path.join(__dirname, ".."), absolutePath);
}

// We need to remove deep calls in the stacktrace and absolute paths;
// otherwise snapshots would be sensitive to context
function normalizeTrace(trace) {
  const inProject = takeUntil(trace, traceObject =>
    /node_modules\/jest/.test(traceObject.fileName)
  );
  return inProject.map(traceObject =>
    Object.assign({}, traceObject, {
      fileName: makePathRelativeToProject(traceObject.fileName),
      evalOrigin: makePathRelativeToProject(traceObject.evalOrigin)
    })
  );
}

// TODO: Actual tests

describe("json format", () => {
  it("matches previous version", () => {
    expect(
      format.json(tree(app), (k, v) => (k === "trace" ? normalizeTrace(v) : v))
    ).toMatchSnapshot();
  });
});

describe("compact format", () => {
  it("matches previous version", () => {
    expect(format.compact(tree(app))).toMatchSnapshot();
  });
});
