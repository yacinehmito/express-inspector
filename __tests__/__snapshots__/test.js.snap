// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compact format matches previous version 1`] = `
"Route /foos                [2m__tests__/_app.js:5:5[0m
  POST <anonymous>         [2m__tests__/_app.js:5:5[0m
Route /foos/:id            [2m__tests__/_app.js:17:1[0m
  GET <anonymous>          [2m__tests__/_app.js:18:1[0m
  DELETE <anonymous>       [2m__tests__/_app.js:21:7[0m
Router /bars               [2m__tests__/_app.js:35:5[0m
  Route /                  [2m__tests__/_app.js:27:8[0m
    GET <anonymous>        [2m__tests__/_app.js:27:8[0m
  Route /:id               [2m__tests__/_app.js:31:8[0m
    GET <anonymous>        [2m__tests__/_app.js:31:8[0m
"
`;

exports[`json format matches previous version 1`] = `
"{
  \\"type\\": \\"app\\",
  \\"instance\\": \\"[Function]\\",
  \\"children\\": [
    {
      \\"type\\": \\"handler\\",
      \\"instance\\": \\"[Function]\\",
      \\"name\\": \\"query\\",
      \\"trace\\": [
        {
          \\"typeName\\": \\"Object\\",
          \\"functionName\\": \\"Object.<anonymous>\\",
          \\"methodName\\": null,
          \\"fileName\\": \\"__tests__/_app.js\\",
          \\"lineNumber\\": 5,
          \\"columnNumber\\": 5,
          \\"evalOrigin\\": \\"__tests__/_app.js\\",
          \\"toplevel\\": false,
          \\"eval\\": false,
          \\"native\\": false,
          \\"constructor\\": false
        }
      ]
    },
    {
      \\"type\\": \\"handler\\",
      \\"instance\\": \\"[Function]\\",
      \\"name\\": \\"expressInit\\",
      \\"trace\\": [
        {
          \\"typeName\\": \\"Object\\",
          \\"functionName\\": \\"Object.<anonymous>\\",
          \\"methodName\\": null,
          \\"fileName\\": \\"__tests__/_app.js\\",
          \\"lineNumber\\": 5,
          \\"columnNumber\\": 5,
          \\"evalOrigin\\": \\"__tests__/_app.js\\",
          \\"toplevel\\": false,
          \\"eval\\": false,
          \\"native\\": false,
          \\"constructor\\": false
        }
      ]
    },
    {
      \\"type\\": \\"route\\",
      \\"instance\\": \\"[Route]\\",
      \\"children\\": [
        {
          \\"type\\": \\"handler\\",
          \\"instance\\": \\"[Function]\\",
          \\"name\\": \\"<anonymous>\\",
          \\"trace\\": [
            {
              \\"typeName\\": \\"Object\\",
              \\"functionName\\": \\"Object.<anonymous>\\",
              \\"methodName\\": null,
              \\"fileName\\": \\"__tests__/_app.js\\",
              \\"lineNumber\\": 5,
              \\"columnNumber\\": 5,
              \\"evalOrigin\\": \\"__tests__/_app.js\\",
              \\"toplevel\\": false,
              \\"eval\\": false,
              \\"native\\": false,
              \\"constructor\\": false
            }
          ],
          \\"method\\": \\"post\\"
        }
      ],
      \\"trace\\": [
        {
          \\"typeName\\": \\"Object\\",
          \\"functionName\\": \\"Object.<anonymous>\\",
          \\"methodName\\": null,
          \\"fileName\\": \\"__tests__/_app.js\\",
          \\"lineNumber\\": 5,
          \\"columnNumber\\": 5,
          \\"evalOrigin\\": \\"__tests__/_app.js\\",
          \\"toplevel\\": false,
          \\"eval\\": false,
          \\"native\\": false,
          \\"constructor\\": false
        }
      ],
      \\"path\\": \\"/foos\\"
    },
    {
      \\"type\\": \\"handler\\",
      \\"instance\\": \\"[Function]\\",
      \\"name\\": \\"<anonymous>\\",
      \\"trace\\": [
        {
          \\"typeName\\": \\"Object\\",
          \\"functionName\\": \\"Object.<anonymous>\\",
          \\"methodName\\": null,
          \\"fileName\\": \\"__tests__/_app.js\\",
          \\"lineNumber\\": 9,
          \\"columnNumber\\": 5,
          \\"evalOrigin\\": \\"__tests__/_app.js\\",
          \\"toplevel\\": false,
          \\"eval\\": false,
          \\"native\\": false,
          \\"constructor\\": false
        }
      ],
      \\"path\\": \\"/\\"
    },
    {
      \\"type\\": \\"route\\",
      \\"instance\\": \\"[Route]\\",
      \\"children\\": [
        {
          \\"type\\": \\"handler\\",
          \\"instance\\": \\"[Function]\\",
          \\"name\\": \\"<anonymous>\\",
          \\"trace\\": [
            {
              \\"this\\": \\"[Object]\\",
              \\"typeName\\": \\"Object\\",
              \\"function\\": \\"[Function]\\",
              \\"functionName\\": \\"Object.<anonymous>\\",
              \\"methodName\\": null,
              \\"fileName\\": \\"__tests__/_app.js\\",
              \\"lineNumber\\": 18,
              \\"columnNumber\\": 1,
              \\"evalOrigin\\": \\"__tests__/_app.js\\",
              \\"toplevel\\": false,
              \\"eval\\": false,
              \\"native\\": false,
              \\"constructor\\": false
            }
          ],
          \\"method\\": \\"get\\"
        },
        {
          \\"type\\": \\"handler\\",
          \\"instance\\": \\"[Function]\\",
          \\"name\\": \\"<anonymous>\\",
          \\"trace\\": [
            {
              \\"this\\": \\"[Object]\\",
              \\"typeName\\": \\"Object\\",
              \\"function\\": \\"[Function]\\",
              \\"functionName\\": \\"Object.<anonymous>\\",
              \\"methodName\\": null,
              \\"fileName\\": \\"__tests__/_app.js\\",
              \\"lineNumber\\": 21,
              \\"columnNumber\\": 7,
              \\"evalOrigin\\": \\"__tests__/_app.js\\",
              \\"toplevel\\": false,
              \\"eval\\": false,
              \\"native\\": false,
              \\"constructor\\": false
            }
          ],
          \\"method\\": \\"delete\\"
        }
      ],
      \\"trace\\": [
        {
          \\"typeName\\": \\"Object\\",
          \\"functionName\\": \\"Object.<anonymous>\\",
          \\"methodName\\": null,
          \\"fileName\\": \\"__tests__/_app.js\\",
          \\"lineNumber\\": 17,
          \\"columnNumber\\": 1,
          \\"evalOrigin\\": \\"__tests__/_app.js\\",
          \\"toplevel\\": false,
          \\"eval\\": false,
          \\"native\\": false,
          \\"constructor\\": false
        }
      ],
      \\"path\\": \\"/foos/:id\\"
    },
    {
      \\"type\\": \\"router\\",
      \\"instance\\": \\"[Function]\\",
      \\"children\\": [
        {
          \\"type\\": \\"route\\",
          \\"instance\\": \\"[Route]\\",
          \\"children\\": [
            {
              \\"type\\": \\"handler\\",
              \\"instance\\": \\"[Function]\\",
              \\"name\\": \\"<anonymous>\\",
              \\"trace\\": [
                {
                  \\"typeName\\": \\"Object\\",
                  \\"functionName\\": \\"Object.<anonymous>\\",
                  \\"methodName\\": null,
                  \\"fileName\\": \\"__tests__/_app.js\\",
                  \\"lineNumber\\": 27,
                  \\"columnNumber\\": 8,
                  \\"evalOrigin\\": \\"__tests__/_app.js\\",
                  \\"toplevel\\": false,
                  \\"eval\\": false,
                  \\"native\\": false,
                  \\"constructor\\": false
                }
              ],
              \\"method\\": \\"get\\"
            }
          ],
          \\"trace\\": [
            {
              \\"typeName\\": \\"Object\\",
              \\"functionName\\": \\"Object.<anonymous>\\",
              \\"methodName\\": null,
              \\"fileName\\": \\"__tests__/_app.js\\",
              \\"lineNumber\\": 27,
              \\"columnNumber\\": 8,
              \\"evalOrigin\\": \\"__tests__/_app.js\\",
              \\"toplevel\\": false,
              \\"eval\\": false,
              \\"native\\": false,
              \\"constructor\\": false
            }
          ],
          \\"path\\": \\"/\\"
        },
        {
          \\"type\\": \\"route\\",
          \\"instance\\": \\"[Route]\\",
          \\"children\\": [
            {
              \\"type\\": \\"handler\\",
              \\"instance\\": \\"[Function]\\",
              \\"name\\": \\"<anonymous>\\",
              \\"trace\\": [
                {
                  \\"typeName\\": \\"Object\\",
                  \\"functionName\\": \\"Object.<anonymous>\\",
                  \\"methodName\\": null,
                  \\"fileName\\": \\"__tests__/_app.js\\",
                  \\"lineNumber\\": 31,
                  \\"columnNumber\\": 8,
                  \\"evalOrigin\\": \\"__tests__/_app.js\\",
                  \\"toplevel\\": false,
                  \\"eval\\": false,
                  \\"native\\": false,
                  \\"constructor\\": false
                }
              ],
              \\"method\\": \\"get\\"
            }
          ],
          \\"trace\\": [
            {
              \\"typeName\\": \\"Object\\",
              \\"functionName\\": \\"Object.<anonymous>\\",
              \\"methodName\\": null,
              \\"fileName\\": \\"__tests__/_app.js\\",
              \\"lineNumber\\": 31,
              \\"columnNumber\\": 8,
              \\"evalOrigin\\": \\"__tests__/_app.js\\",
              \\"toplevel\\": false,
              \\"eval\\": false,
              \\"native\\": false,
              \\"constructor\\": false
            }
          ],
          \\"path\\": \\"/:id\\"
        }
      ],
      \\"trace\\": [
        {
          \\"typeName\\": \\"Object\\",
          \\"functionName\\": \\"Object.<anonymous>\\",
          \\"methodName\\": null,
          \\"fileName\\": \\"__tests__/_app.js\\",
          \\"lineNumber\\": 35,
          \\"columnNumber\\": 5,
          \\"evalOrigin\\": \\"__tests__/_app.js\\",
          \\"toplevel\\": false,
          \\"eval\\": false,
          \\"native\\": false,
          \\"constructor\\": false
        }
      ],
      \\"path\\": \\"/bars\\"
    }
  ]
}"
`;
